src/html_extractor.py

```diff
@@ -0,0 +1,941 @@
+			
+		</div>
+		
+	</div> <!-- end: #main -->
+	<div id="footer">
+		
+
+
+
+<div style="min-height: 15px;">
+	<ul>
+		<li>RK-Termin&nbsp;1.2.36.1</li>
+		<li style="margin-left: 5em;">
+			    
+				<a href="extern/dsgvo.do?request_locale=ru" target="_blank" title="Информация о защите данных и правила пользования"><img src="images/flags/ru.png" alt="Информация о защите данных и правила пользования" title="Информация о защите данных и правила пользования">&nbsp; Информация
+					о защите данных и правила пользования</a>
+			 </li>
+	</ul>
+</div>
+
+
+
+	</div>
+</div>
+
+<div id="ui-datepicker-div" class="ui-datepicker ui-widget ui-widget-content ui-helper-clearfix ui-corner-all"></div><div class="ui-dialog ui-widget ui-widget-content ui-corner-all ui-front ui-draggable" tabindex="-1" role="dialog" aria-describedby="commit-request" style="display: none;" aria-labelledby="ui-id-1"><div class="ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix ui-draggable-handle"><span id="ui-id-1" class="ui-dialog-title">В обработке ... </span><button type="button" class="ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only ui-dialog-titlebar-close" role="button" title="Close"><span class="ui-button-icon-primary ui-icon ui-icon-closethick"></span><span class="ui-button-text">Close</span></button></div><div id="commit-request" class="ui-dialog-content ui-widget-content">
+	<p>
+		<span class="ui-icon ui-icon-alert" style="float: left; margin: 0 7px 20px 0; margin-bottom: 100px"></span>
+		<span id="question">
+			<b>подождите ...</b><br>Ваш запрос обрабатывается. Пожалуйста, дождитесь ответа. Перезагрузка страницы замедлит обработку.
+		</span>
+		
+	</p>
+	<div id="commit-request-progressbar" style="background: #7192b6;" class="ui-progressbar ui-widget ui-widget-content ui-corner-all ui-progressbar-indeterminate" role="progressbar" aria-valuemin="0">
+	
+		<div style="background: rgba(0,0,0,0) url('images/indet.gif'); opacity:0.25; height:2em;">
+			&nbsp;
+		</div>
+	
+	<div class="ui-progressbar-value ui-widget-header ui-corner-left" style="width: 100%;"><div class="ui-progressbar-overlay"></div></div></div>
+</div></div></body></html>
+"""
+
+only_captcha_tags = SoupStrainer("captcha")
+only_a_tags = SoupStrainer("a")
+
+
+def extract_captcha(html):
+    soup = BeautifulSoup(html, 'html.parser', parse_only=only_captcha_tags)
+    result = soup.captcha.contents[1].get('style')[44:-78]
+    return result
+
+
+def extract_day_href(html):
+    soup = BeautifulSoup(html, 'html.parser', parse_only=only_a_tags)
+    result = soup.find("a", "arrow").get("href")
+    return result
+
+
+def extract_time_slot_href(html):
+    soup = BeautifulSoup(html, 'html.parser', parse_only=only_a_tags)
+    result = soup.find("a", string="Записаться на прием").get('href')
+    return result
+
+# print(extract_captcha(html_doc_with_captcha))
+# print(extract_day_href(html_doc_with_day))
+# print(extract_time_slot_href(html_doc_with_time_slot))
```

Comment:
Extract these prints as 3 different tests in tests/test_extractor/test_extractor.py
Place test content (lines 0-916) into a separate file in tests/test_extractor/page_example.txt